### PR TITLE
update ci-signal reporter OWNERS

### PR DESCRIPTION
**Pull request by leonardpahlke**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

This PR updates the OWNERS file in the ci-signal report tool. It also deletes the OWNERS_ALIAS reference which has been created for that. 

The entire ci-signal team has been added as reviewer to the sub-project (k/release/cmd/ci-reporter). From my perspective, nobody on that lists fulfills the requirements that be a reviewer for this sub-project ([requirements to be a reviewer](https://github.com/kubernetes/community/blob/master/community-membership.md#reviewer)). I thinks this has been mistakenly added initially [see PR](https://github.com/kubernetes/release/pull/2309)

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

None

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

/cc @encodeflush @rajula96reddy @simrangupta234 @RinkiyaKeDad @calvh 

/assign @justaugustus @puerco

### CODE REVIEW

**Review by saschagrunert — APPROVED** [review]

**Comment by justaugustus** [comment]
> /hold for review

@leonardpahlke -- For clarity, I initially put a hold on this to gather feedback from @kubernetes/sig-release-admins.

> I thinks this has been mistakenly added initially [see PR](https://github.com/kubernetes/release/pull/2309)

The aliases were intentionally added.
See my review here: https://github.com/kubernetes/release/pull/2309#pullrequestreview-801463833

> The entire ci-signal team has been added as reviewer to the sub-project (k/release/cmd/ci-reporter). From my perspective, nobody on that lists fulfills the requirements that be a reviewer for this sub-project ([requirements to be a reviewer](https://github.com/kubernetes/community/blob/master/community-membership.md#reviewer)).

Unfortunately, I think the above linked document stands as guidance for various areas of the project, not a true representation of something you'll see consistent across subprojects.

For SIG Release and @kubernetes/release-engineering in particular, there are a few areas that we may grant contributors ownership/access to with the expectation that they can/will grow into that responsibility.

Examples:
- blanket org membership for @kubernetes/release-team members
- delegating changelog approvals to the Release Notes subteam of the Release Team: https://github.com/kubernetes/kubernetes/pull/104759

In the latter instance, the expectation is for the Release Notes subteam to act as a reviewers, while the Release Notes Lead for the cycle has approval rights.
This federates the load of say, reviewing typo fixes, to groups outside of core Release Engineering.

I strongly believe this is another area where we can delegate.
Expectations:
- CI Signal team reviews improvements to the CI reporting tooling
- Core Release Engineering approves

This improvement doesn't obviate the need for Release Engineering approval, but it does provide a mechanism by which CI Signal subteam members can attain review access extending beyond any one release cycle.

The alternative is for CI Signal folks to meticulously watch this repo (maybe even a new label therein) and do reviews on some cadence.
But!
1. RelEng members are likely to beat newer reviewers to the review because we're getting actively assigned
2. Why not let the bots handle it?

Happy to discuss in more detail, if needed here/channels/meetings/elsewhere, but @kubernetes/sig-release-admins have decided to close this as not required.

/close

**Comment by puerco** [comment]
Agree. As the tool is now under SIG Release, we should update the CI Signal handbooks to add the new expectations and optimal requirements for the team. I've opened the following issue to track the changes to the handbooks:

https://github.com/kubernetes/sig-release/issues/1755

**Pull request closed without merging**
